CanvasColorSpaceProposal.md

```diff
@@ -63,56 +89,58 @@ partial interface CanvasRenderingContext2D {
-   * This color space has a linear transfer function that supports arbitrary pixel values, in particular, values outside of the range [0, 1].
-* Note that gradients will be physically uniform, and thus will not match gradients in an "srgb" canvas
-* Note that pixel values written to WebGL draw buffers are now in a space with a linear transfer function, and will not necessarily match those of an "srgb" canvas.
-
-#### The pixelFormat context creation attribute
-The pixelFormat attributes specifies the numeric types to be used for storing pixel values.
-* Support for "8-8-8-8" is mandatory. All other formats ar optional.
-* When an unsupported format is requested, the format shall fall back to "8-8-8-8".
-* With formats that use integer numeric types for the color channels, the canvas pixel buffer shall store non-linear color values, encoded using the transfer curves prescribed by the specification of the current color space.
-* With formats that use floating-point numeric types for the color channels, the canvas pixel buffer shall store linear (i.e unencoded) color values.
-* Float values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a pixel value of (2,2,2) should be twice as bright as (1,1,1).
+* toDataURL/toBlob produce resources tagged as being in the sRGB color space, if the encoding format supports colorspace tagging or embedded color profiles.
+    * Exception: User agents that implement the legacy behavior must not encode any color space metadata.
+
+##### The "rec-2020" color space
+* As per the [CSS rec-2020 color space](https://www.w3.org/TR/css-color-4/#valdef-color-rec-2020).
+* This color space has different primaries and a different transfer function than "srgb".
+* Support is optional, and should not be present on srgb-only UAs.
+
+
+#### The colorEncoding context creation attribute
+The colorEncoding attributes specifies the encoding to be used for storing pixel channel color values.
+* Support for "unorm8" is mandatory. All other encodings are optional.
+* When an unsupported encoding is requested, the encoding shall fall back to "unorm8".
+* The alpha channel is always interpreted as if clamped to [0,1].
+* Float RGB channel values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a (luminance-linear) pixel value of (2,2,2) should be twice as bright as (1,1,1).
```

Comment:
I think we should clamp to [0, 1] except for color spaces (e.g, "extended-srgb") which explicitly support values outside the range of [0, 1].

The reason for this is that the extended values can be used for HDR, and enabling HDR often requires falling down a different (higher power) path.